crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -213,7 +219,7 @@ impl BeaconConsensusEngineHandle {
 ///
 /// If the future is polled more than once. Leads to undefined state.
 #[must_use = "Future does nothing unless polled"]
-pub struct BeaconConsensusEngine<DB, BT, Client>
+pub struct BeaconConsensusEngine<DB, BT, Client, St>
```

Comment:
I wanted to move the CanonStateNotification out of the blockchain tree anyway and let the beacon engine do the notifying.

So we can keep this, otherwise I'd prefer if the Pruner could just box the stream, perhaps it should anyways? just to make it easier to use, the type only fires an event every 12s so there's no perf to gain/lose with dyn